File: Data Structures & Algorithms/binary-search/submission-0.js
class Solution {
    /**
     * @param {number[]} nums
     * @param {number} target
     * @return {number}
     */
    search(nums, target) {
        let left = 0;
        let right = nums.length - 1;

        while (left <= right) {
            let middle = Math.floor((left + right) / 2);

            if (nums[middle] === target) return middle;

            if (nums[middle] > target) {
                // Too large, move left
                right = middle - 1;
            } else if (nums[middle] < target) {
                // Too small, move right
                left = middle + 1;
            }
        }
        return -1;
    }
}

File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-0.py
class Solution:
    def findMin(self, nums: List[int]) -> int:
        left = 0
        right = len(nums) - 1

        # Searching for a specific number, so no =
        while (left < right):
            middle = (left + right) // 2

            if nums[middle] > nums[right]:
                # Smaller numbers are to the right, move right
                left = middle + 1
            else:
                # Middle could be the smallest
                right = middle

        return nums[right]